jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -214,26 +245,56 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
             nddata = spectral_cube.get_subset_object(
                 subset_id=self.aperture.selected, cls=NDDataArray
             )
-            uncertainties = uncert_cube.get_subset_object(
-                subset_id=self.aperture.selected, cls=StdDevUncertainty
+            if uncert_cube:
+                uncertainties = uncert_cube.get_subset_object(
+                    subset_id=self.aperture.selected, cls=StdDevUncertainty
+                )
+            # Exact slice mask of cone or cylindrical aperture through the cube. `shape_mask` is
+            # a 3D array with fractions of each pixel within an aperture at each
+            # wavelength, on the range [0, 1].
+            if self.function_selected.lower() in ['min', 'max'] and self.wavelength_dependent:
+                shape_mask = self.cone_aperture().astype(int)
```

Comment:
I thought this combo isn't allowed? Shouldn't this have thrown an error earlier?

Reply:
Oh wait, nvm, I confused myself.